src/common/utils/formatter/formatter.ts

```diff
@@ -91,18 +92,29 @@ function getFormattedStringClauseValues({ values, action }: StringFilterClause):
   }
 }
 
+function getFormattedBooleanClauseValues({ values }: BooleanFilterClause) {
+  return values.count() > 1 ? `(${values.count()})` : values.first().toString();
+}
+
+function getFormattedNumberClauseValues(clause: NumberFilterClause) {
+  const { start, end } = clause.values.first();
+  return `${start} to ${end}`;
+}
+
 function getFilterClauseValues(clause: FilterClause, timezone: Timezone): string {
```

Comment:
Could we use polymorphic calls here if the  move getFormattedTimeClauseValues to the clause class? But how to handle timezone for time clause?

Reply:
In perfect case I would define `getFilterClauseValues` as trait/typeclass and define implementations outside of `FilterClause`. There is a lot of "formatter" logic inside and I think it belongs there.